R/get_test_data.R

```diff
@@ -21,22 +21,19 @@
 #'  get_test_data(recipe = rec, x = case_death_rate_subset)
 #' @export
 
-get_test_data <- function(recipe, x){
-  # TO-DO: SOME CHECKS OF THE DATASET
-  if (any(!(c('geo_value','time_value') %in% colnames(x)))) {
-    rlang::abort("`geo_value`, `time_value` does not exist in data")
-  }
-  ## CHECK if it is epi_df?
-
-  max_lags <- max(map_dbl(recipe$steps, ~ max(.x$lag %||% 0)))
+get_test_data <- function(recipe, x) {
+  stopifnot(is.data.frame(x))
+  if (! all(colnames(x) %in% colnames(recipe$template)))
+    cli_stop("some variables used for training are not available in `x`.")
+  max_lags <- max(map_dbl(recipe$steps, ~ max(.x$lag %||% 0)), 0)
 
   # CHECK: Return NA if insufficient training data
   if (dplyr::n_distinct(x$time_value) < max_lags) {
     cli_stop("You supplied insufficient training data for this recipe. ",
              "You need at least {max_lags} distinct time_values.")
   }
 
-  groups <- epi_keys(recipe)[epi_keys(recipe) != "time_value"]
+  groups <- epi_keys(recipe)[-1]
```

Comment:
It's not immediately clear to me what this line is trying to do. I think it would be helpful to have a comment here to explain "exclude time_value", or "epi_keys will return with time_value as the first element"

Side note, I totally thought `[-1]` was grabbing the last element of the vector until I looked it up

Reply:
The assumption that `time_value` is the 1st element is fair at this time, but it's probably best to just use the col name there if possible because if at some point in the future we choose not to enforce col. ordering, this could be problem.

Reply:
Agreed. Fixing.